runtime/fiber.c

```diff
@@ -102,10 +102,10 @@ struct stack_info** caml_alloc_stack_cache (void)
 
 Caml_inline struct stack_info* alloc_for_stack (mlsize_t wosize)
 {
-  size_t len = sizeof(struct stack_info) +
-               sizeof(value) * wosize +
-               8 /* for alignment to 16-bytes, needed for arm64 */ +
-               sizeof(struct stack_handler);
+  size_t len =
+    /* alignment to 16-bytes is needed at least for arm64 */
+    caml_round_up(sizeof(struct stack_info) + sizeof(value) * wosize, 16) +
+    sizeof(struct stack_handler);
```

Comment:
Shouldn't the `sizeof(struct stack_handler)` part be included as a `caml_round_up` argument, instead of being added afterwards?

Reply:
No - you want the `stack_handler` itself to be aligned to the required boundary.